interp/src/interpreter.rs

```diff
@@ -0,0 +1,619 @@
+
+                // update internal state of the cell and
+                // queue any required updates.
+
+                //determine if dst_cell is a combinational cell or not
+                if get_combinational_or_not(
+                    &dst_cell,
+                    &assign.dst.borrow().name,
+                    env,
+                ) {
+                    // write to assign.dst to e2 immediately, if combinational
+                    write_env.put(
+                        &dst_cell,
+                        &assign.dst.borrow().name,
+                        read_val,
+                    );
+
+                    /*println!(
+                        "reg0.write_en = {}",
+                        write_env.get(
+                            &ir::Id::from("reg0"),
+                            &ir::Id::from("write_en")
+                        )
+                    );*/
+
+                    // now, update the internal state of the cell; for now, this only includes adds; TODO (use primitive Cell parameters)
+                    let inputs;
+                    let outputs;
+
+                    // TODO: hacky way to avoid updating the cell state. Also, how to get input and output vectors in general??
+                    if &assign.dst.borrow().name != "write_en" {
+                        // get dst_cell's input vector
+                        match &write_env.get_cell(&dst_cell) {
+                            Some(cell) => {
+                                inputs = vec![
+                                    (cell.borrow())
+                                        .get("left")
+                                        .borrow()
+                                        .name
+                                        .clone(),
+                                    (cell.borrow())
+                                        .get("right")
+                                        .borrow()
+                                        .name
+                                        .clone(),
+                                ]
+                            }
+                            _ => panic!("could not find cell"),
+                        }
+
+                        // get dst_cell's output vector
+                        match &write_env.get_cell(&dst_cell) {
+                            Some(cell) => {
+                                outputs = vec![(cell.borrow())
+                                    .get("out")
+                                    .borrow()
+                                    .name
+                                    .clone()]
+                                //clean this up later?
+                            }
+                            _ => panic!("could not find cell"),
+                        }
+
+                        // update the cell state in write_env
+                        write_env = update_cell_state(
+                            &dst_cell,
+                            &inputs[..],
```

Comment:
don't need `[..]`. just `&inputs` is enough.

Reply:
Changed